src/connectors/BitKeepConnector/index.ts

```diff
@@ -0,0 +1,124 @@
+import { AbstractConnectorArguments, ConnectorUpdate } from '@web3-react/types';
+import warning from 'tiny-warning';
+import { NoEthereumProviderError, UserRejectedRequestError } from '../DefiConnector';
+
+export class BitKeepConnector extends AbstractConnector {
+  constructor(kwargs: AbstractConnectorArguments) {
+    super(kwargs);
+
+    this.handleNetworkChanged = this.handleNetworkChanged.bind(this);
+    this.handleChainChanged = this.handleChainChanged.bind(this);
+    this.handleAccountsChanged = this.handleAccountsChanged.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+  }
+
+  private handleChainChanged(chainId: string | number): void {
+    console.log("Handling 'chainChanged' event with payload", chainId);
+    this.emitUpdate({ chainId, provider: window?.bitkeep?.ethereum });
+  }
+
+  private handleAccountsChanged(accounts: string[]): void {
+    console.log("Handling 'accountsChanged' event with payload", accounts);
+    if (accounts.length === 0) {
+      this.emitDeactivate();
+    } else {
+      this.emitUpdate({ account: accounts[0] });
+    }
+  }
+
+  private handleClose(code: number, reason: string): void {
+    console.log("Handling 'close' event with payload", code, reason);
+    this.emitDeactivate();
+  }
+
+  private handleNetworkChanged(networkId: string | number): void {
+    console.log("Handling 'networkChanged' event with payload", networkId);
+    this.emitUpdate({ chainId: networkId, provider: window?.bitkeep?.ethereum });
+  }
+
+  public async activate(): Promise<ConnectorUpdate> {
+    if (!window?.bitkeep?.ethereum) {
+      throw new NoEthereumProviderError();
+    }
+
+    if (window?.bitkeep?.ethereum.on) {
+      window.bitkeep.ethereum.on('chainChanged', this.handleChainChanged);
+      window.bitkeep.ethereum.on('accountsChanged', this.handleAccountsChanged);
+      window.bitkeep.ethereum.on('close', this.handleClose);
+      window.bitkeep.ethereum.on('networkChanged', this.handleNetworkChanged);
+    }
+
+    if (window?.bitkeep?.ethereum.isBitKeep) {
+      window.bitkeep.ethereum.autoRefreshOnNetworkChange = false;
+    }
+
+    // try to activate + get account via eth_requestAccounts
+    let account: string | undefined;
+    try {
+      const accounts: string[] | undefined = await window.bitkeep?.ethereum.request({ method: 'eth_requestAccounts' });
+      if (accounts && accounts.length) account = accounts[0];
+    } catch (error) {
+      if ((error as any).code === 4001) {
+        throw new UserRejectedRequestError();
+      } else {
+        warning(false, 'eth_requestAccounts was unsuccessful, falling back to enable');
+      }
+    }
+
+    return { provider: window.bitkeep.ethereum, account };
+  }
+
+  public async getProvider() {
+    const provider = window.bitkeep && window.bitkeep.ethereum;
+    if (!provider) {
+      return window.open('https://bitkeep.com/download?type=0&theme=light');
+    }
```

Comment:
I think we don't need this link here. because it will automatically redirect without user consent. We already have Install BitKeep button so that is fine.